internal/daemon/vendor-specific-plugins/marvell/main.go

```diff
@@ -255,29 +360,54 @@ func (vsp *mrvlVspServer) GetDevices(ctx context.Context, in *pb.Empty) (*pb.Dev
 	}, nil
 }
 
+// SetNumVfs function to set the number of VFs with the given context and VfCount
+func (vsp *mrvlVspServer) SetNumVfs(ctx context.Context, in *pb.VfCount) (*pb.VfCount, error) {
+	vsp.log.Info("Received SetNumVfs() request", "VfCnt", in.VfCnt)
+	if vsp.isDPUMode {
+		return nil, errors.New("SetNumVfs is not supported in DPU Mode")
+	}
+	pciAddress, err := mrvlutils.GetPCIByDeviceID(HostDeviceID)
+	if pciAddress == "" || err != nil {
+		return nil, errors.New("PCI Address not found")
+	}
+	vfcnt := in.VfCnt
+	if vfcnt < 0 {
+		return nil, errors.New("invalid VF Count")
+	}
+	cmd := exec.Command("sh", "-c", fmt.Sprintf("echo %d > /sys/bus/pci/devices/0000:17:00.0/sriov_numvfs", vfcnt))
```

Comment:
doesn't somebody need to take care that if sriov_numvfs is not zero, that it first gets reset to zero? Who does that or how is this supposed to work?